pkg/compactor/bucket_compactor.go

```diff
@@ -361,6 +363,11 @@ func (c *BucketCompactor) runCompactionJob(ctx context.Context, job *Job) (shoul
 	uploadBegin := time.Now()
 	uploadedBlocks := atomic.NewInt64(0)
 
+	if err = verifyCompactedBlocksTimeRanges(compIDs, toCompactMinTime.UnixMilli(), toCompactMaxTime.UnixMilli(), subDir); err != nil {
+		level.Error(jobLogger).Log("msg", err)
```

Comment:
[nit] I suggest to add a common "msg" which is easier to search in logs, and then the details in the "err" field.

```suggestion
		level.Error(jobLogger).Log("msg", "compacted blocks verification failed", "err", err)
```